Purchases/Attribution/RCAttributionFetcher.m

```diff
@@ -137,15 +137,32 @@ - (BOOL)isAuthorizedToPostSearchAds {
 
         BOOL needsTrackingAuthorization = [self.systemInfo isOperatingSystemAtLeastVersion:minimumOSVersionRequiringAuthorization];
 
-        Class<FakeATTrackingManager> _Nullable trackingManagerClass = [self.attributionFactory atTrackingManagerClass];
-        if (!trackingManagerClass && needsTrackingAuthorization) {
-            RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+        Class _Nullable trackingManagerClass = [self.attributionFactory atTrackingClass];
+        if (!trackingManagerClass) {
+            if (needsTrackingAuthorization) {
+                RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+            }
+            return !needsTrackingAuthorization;
+        }
+        SEL authStatusSelector = NSSelectorFromString(self.attributionFactory.authorizationStatusPropertyName);
+        BOOL canPerformSelector = [trackingManagerClass respondsToSelector:authStatusSelector];
+        if (!canPerformSelector) {
```

Comment:
we first check that we _can_ call the method, although we should be able to really